### PR TITLE
Enable sonarqube scanning for frontend code

### PR DESCRIPTION
**Pull request by defaude**
Relates to #263

* update a bunch of test-related dependencies
* add docker-capable "headless chrome" config for karma
* add "npm run test:ci" script for running tests single-run, no-watch, but with coverage
* configure sonar project properties
* add "npm run sonar" script to perform sonar analysis

I don't precisely know how the project is configured over at sonarcloud.io - or whether that's the place where we're supposed to run the analysis. However, updating the `sonar-project.properties` file and adding the `npm run test:ci && npm run sonar` to the automated build should be a no-brainer once the sonar server is set up correctly.

### CODE REVIEW

**Comment by vlow** [comment]
@defaude Thanks for the request.

The sonarcloud.io analysis is broken atm: the qaware organization on sonarcloud is no longer the owner of this project, since we migrated to a dedicated lokaler-kaufen organization on github. I just created a new organization over at sonarcloud.io and updated the values in the build.gradle file. @fuchshuber is installing the new SONAR_TOKEN in our Google Cloud Build, so things should be working today EOB.

Can you please adjust the configuration values in your sonar-project properties to those used in build.gradle and update your pull request?

```
        property "sonar.projectKey", "lokaler-kaufen_lokaler-kaufen-app"
        property "sonar.organization", "lokaler-kaufen"
        property "sonar.host.url", "https://sonarcloud.io"
```

File: frontend/sonar-project.properties [hunk 1]
```diff
@@ -0,0 +1,12 @@
+sonar.projectKey=qaware_wir-vs-virus_frontend
```
Comment (vlow) [inline frontend/sonar-project.properties:1]: Bitte auf neue Org umziehen (siehe Kommentar an #270)

File: frontend/sonar-project.properties [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+sonar.projectName=QAware Wir vs. Virus (Frontend)
+sonar.organization=qaware
```
Comment (vlow) [inline frontend/sonar-project.properties:3]: Hier bitte auch an die neue Org anpassen.

**Comment by defaude** [comment]
Hab die Config entsprechend angepasst.

**Pull request merged**